src/app/3d/qgs3dmapconfigwidget.h

```diff
@@ -48,6 +48,8 @@ class Qgs3DMapConfigWidget : public QWidget, private Ui::Map3DConfigWidget
     void onTerrainLayerChanged();
     void updateMaxZoomLevel();
     void validate();
+    void update3dAxisMode();
```

Comment:
I think it would be better if you have both functions as one and make a function named something like update3Daxis that checks for which configuration has changed between UI and the object and makes changes